numba/tests/test_np_functions.py

```diff
@@ -1092,6 +1092,30 @@ def test_partition_boolean_inputs(self):
             for kth in True, False, -1, 0, 1:
                 self.partition_sanity_check(pyfunc, cfunc, d, kth)
 
+    @unittest.skipUnless(np_version >= (1, 10), "cov needs Numpy 1.10+")
+    @needs_blas
+    def test_cov_invalid_ddof(self):
+        pyfunc = cov
+        cfunc = jit(nopython=True)(pyfunc)
+
+        # Exceptions leak references
+        self.disable_leak_check()
+
+        def _check(m, ddof, expected_msg):
+            with self.assertTypingError() as raises:
+               cfunc(m, ddof=ddof)
+            self.assertIn(expected_msg, str(raises.exception))
+
+        m = np.array([[0, 2], [1, 1], [2, 0]]).T
+
+        for ddof in np.nan, -np.inf:
+            msg = 'Cannot convert non-finite ddof to integer'
+            _check(m, ddof, msg)
+
+        for ddof in 'bacon', np.arange(4), 1.1, -3.142:
```

Comment:
Nit: Perhaps s/bacon/junk/ as it's a more common nonsense input value?

Reply:
Yeah, removed it - although I am always keen to include breakfast nouns in tests :)